fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/SocialMediaRepository.kt

```diff
@@ -171,6 +179,11 @@ constructor(appDatabase: AppDatabase, application: Application) {
     private val chirps = ConcurrentHashMap<MessageID, Chirp>()
     private val chirpSubjects = ConcurrentHashMap<MessageID, Subject<Chirp>>()
     private val chirpsSubject: Subject<Set<MessageID>> = BehaviorSubject.createDefault(emptySet())
+    // TODO : used to fake page loading, to be removed when we will be able to request only wanted
+    // chirps. Maxime Teuber @Kaz-ookid - April 2024
+    private var chirpsLoaded: AtomicInteger = AtomicInteger(0)
+    private var storageIsLoaded: Boolean = false
+    private var reachedEndOfChirps: Boolean = false
```

Comment:
This seems a shared variable to me, at least theoretically. As they're accessed in the add method and when data is loaded from disk, which can in theory happen in parallel (usually it never happens as loading from storage is quite fast but I'd still use atomic variables)